imscjs-demo/js/index.js

```diff
@@ -1,59 +1,52 @@
-
-function initTTML() {
-  var renderDiv = document.getElementById("render-div");
-  var myVideo = document.getElementById("imscVideo");
+async function initTTML() {
+  const renderDiv = document.getElementById("render-div");
+  const myVideo = document.getElementById("imscVideo");
   //Track object generated by the first <track> child element
-  var myTrack = myVideo.textTracks[0];
+  const myTrack = myVideo.textTracks[0];
   //Src is only available from the <track> DOM element, not the obj
-  var ttmlUrl = myVideo.getElementsByTagName("track")[0].src;
+  const ttmlUrl = myVideo.getElementsByTagName("track")[0].src;
   //The track is disabled by default, 'hidden' does not display VTT Text
   //but fires the events
   myTrack.mode = "hidden";
 
-  var client = new XMLHttpRequest();
-
-  client.open('GET', ttmlUrl);
-  client.onreadystatechange = function () {
-    initTrack(client.responseText);
-  }
-  client.send();
-
+  const response = await fetch(ttmlUrl);
+  initTrack(await response.text());
 
   function initTrack(text) {
-    var imscDoc = imsc.fromXML(text);
-    var timeEvents = imscDoc.getMediaTimeEvents();
+    const imscDoc = imsc.fromXML(text);
+    const timeEvents = imscDoc.getMediaTimeEvents();
     //create cue per timed event and render isd
-    for (var i = 0; i < timeEvents.length; i++) {
+    for (let i = 0; i < timeEvents.length; i++) {
       //Edge/IE implement "Generic Cue", other browsers VTTCue
-      var Cue = window.VTTCue || window.TextTrackCue;
+      const Cue = window.VTTCue || window.TextTrackCue;
```

Comment:
I don't think this is still needed: VTTCue is unlikely to be supported by Edge > 79. Anyway, `TextTrackCue` has been supported by every engine for ages; we can simplify this. (as well as the call to the `Cue()` that would be a `new TextTrackCue()`on line 25 and 30.
![Capture d’écran 2023-12-20 à 06 42 49](https://github.com/mdn/imsc-examples/assets/1466293/44f38979-a95c-4387-bee4-bcad9ba3eb91)

Reply:
As far as I can tell the correct thing here is to use `VTTCue`: it's supported everywhere. `TextTrackCue` is an abstract class, we couldn't create it.